qa/integration-tests/src/test/java/io/camunda/it/client/ProcessQueryTest.java

```diff
@@ -0,0 +1,558 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.it.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.qa.util.cluster.TestStandaloneCamunda;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.command.CreateProcessInstanceCommandStep1;
+import io.camunda.zeebe.client.api.response.*;
+import io.camunda.zeebe.client.api.response.Process;
+import io.camunda.zeebe.client.api.search.filter.ProcessInstanceFilter;
+import io.camunda.zeebe.client.api.search.response.ProcessInstance;
+import io.camunda.zeebe.client.protocol.rest.ProcessInstanceVariableFilterRequest;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration.TestZeebe;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ZeebeIntegration
+public class ProcessQueryTest {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ProcessQueryTest.class);
+
+  private static final List<Process> DEPLOYED_PROCESSES = new ArrayList<>();
+  private static final List<ProcessInstanceEvent> PROCESS_INSTANCES = new ArrayList<>();
+
+  @TestZeebe
+  private static TestStandaloneCamunda testStandaloneCamunda = new TestStandaloneCamunda();
+
+  private static ZeebeClient zeebeClient;
+
+  @BeforeAll
+  public static void setup() throws InterruptedException {
+
+    zeebeClient = testStandaloneCamunda.newClientBuilder().build();
+
+    final List<String> processes =
+        List.of(
+            "service_tasks_v1.bpmn",
+            "service_tasks_v2.bpmn",
+            "incident_process_v1.bpmn",
+            "manual_process.bpmn",
+            "parent_process_v1.bpmn",
+            "child_process_v1.bpmn");
+    processes.forEach(
+        process ->
+            DEPLOYED_PROCESSES.addAll(
+                deployResource(String.format("process/%s", process)).getProcesses()));
+				
```

Comment:
Checkstyle is complaining about this line, there must be some whitespaces to remove